templates/system/modules/pwa/themes/base/img/browserconfig.xml

```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<browserconfig><msapplication><tile><square70x70logo src="{% PH7_LAYOUT . PH7_SYS . PH7_MOD . 'pwa/' . PH7_TPL . PH7_DEFAULT_THEME . PH7_SH . PH7_IMG %}ms-icon-70x70.png"/><square150x150logo src="{% PH7_LAYOUT . PH7_SYS . PH7_MOD . 'pwa/' . PH7_TPL . PH7_DEFAULT_THEME . PH7_SH . PH7_IMG %}ms-icon-150x150.png"/><square310x310logo src="{{ $design->url(PH7_LAYOUT . PH7_SYS,'pwa/' ,'themes/' ,'base/' ,'img') }}ms-icon-310x310.png"/><TileColor>#ffffff</TileColor></tile></msapplication></browserconfig>
```

Comment:
I can fix it anyway, but just for your information, those variables won't be parsed by pH7Tpl. 
So, it will be displayed as it is. 

If we want those variables to be translated to real paths, we need to process the same way I did for the manifest.xml file.

It's a bit complex, so, don't worry, I can do it for you.

But, basically, you need to create a function (e.g. called `browserConfig`), in the MainController of "pwa" module (like the one for "manifest" https://github.com/pH7Software/pH7-Social-Dating-CMS/blob/f16b22a6beee29c938f01f3fff4654cf3678c3bb/_protected/app/system/modules/pwa/controllers/MainController.php#L19-L28

Then, add this file (browserconfig.xml) in `_protected/app/system/modules/pwa/views/base/tpl/main`, and rename it to "browserconfig.xml.tpl"

Then, add a rule to the XML routes, to get the proper URL (www.my-website.com/browserconfig.xml) 

So, like the manifest.xml file, https://github.com/pH7Software/pH7-Social-Dating-CMS/blob/6137a377611b876432b3a1c3d99149e6074477b9/_protected/app/configs/routes/en.xml#L10

But here, it will be <route url="browserconfig\.xml" path="system/modules" module="pwa" controller="main" action="browserconfig" />

Reply:
I understand ;) I just saw this on the lighthouse report... Do we need to update this?

![qqq](https://user-images.githubusercontent.com/52648055/79579923-26211580-80c0-11ea-89b2-c6e9b174ae87.PNG)

Reply:
Oh yes, and probably, create a new output function in the MainController of pwa module,

Like the `jsonOutput()`, https://github.com/pH7Software/pH7-Social-Dating-CMS/blob/f16b22a6beee29c938f01f3fff4654cf3678c3bb/_protected/app/system/modules/pwa/controllers/MainController.php#L36
but for `xmlOutput` (and a new contant at the top of the controller for `.xml` extension.

Reply:
3 javascript libraries are out of date and have known vulnerabilities? JS is not my specialty

Reply:
But once again, don't worry about it. I just explained to you, so you know more how it works (otherwise, I don't help you). But I will do it. You don't have to struggle with this. What you did is already amazing!

Reply:
I have done things very similar, thanks for the explanation :)

Reply:
What are your thoughts on the report?

Reply:
Done this :) Hope its okay it was stressful :)